build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```diff
@@ -140,12 +140,17 @@ fun addDependencies() {
 fun addCompileAllTask() {
     tasks.register("compileAll") {
         val compileTasks = project.tasks.matching {
-            it is JavaCompile || it is GroovyCompile
+            (it is JavaCompile || it is GroovyCompile) && !it.isTestCompile()
         }
         dependsOn(compileTasks)
     }
 }
 
+val testCompileTasks
+    get() = TestType.values().map { "compile${it.prefix.toUpperCase()}Test" } + "compileTest"
```

Comment:
This would also exclude compiling the test fixtures. I think we should compile the test fixtures as part of compileAll.

Reply:
PATL!